Servers/UI/OJS.Servers.Ui/ClientApp/src/hooks/submissions/use-submissions-details.tsx

```diff
@@ -26,37 +25,38 @@ interface ISubmissionsDetailsProviderProps extends IHaveChildrenProps {}
 
 const SubmissionsDetailsProvider = ({ children }: ISubmissionsDetailsProviderProps) => {
     const { startLoading, stopLoading } = useLoading();
-    const { state: { isOfficial: isContestParticipationOfficial } } = useCurrentContest();
+    // const { state: { isOfficial: isContestParticipationOfficial } } = useCurrentContest();
     const [ currentSubmissionId, setCurrentSubmissionId ] = useState<number>();
-    const [ currentSubmission, setCurrentSubmission ] = useState<ISubmissionDetailsType>();
-    const [
-        submissionResultsByProblemUrlParams,
-        setSubmissionResultsByProblemUrlParams ] =
-        useState<IGetSubmissionResultsByProblemUrlParams | null>();
+    const [ currentSubmission, setCurrentSubmission ] = useState<ISubmissionDetailsType | null>(null);
     const [ currentProblemSubmissionResults, setCurrentProblemSubmissionResults ] =
         useState<ISubmissionDetails[]>(defaultState.currentProblemSubmissionResults);
 
     const { getSubmissionDetailsByIdUrl, getSubmissionResultsByProblemUrl } = useUrls();
 
+    const [
+        getSubmissionDetailsByIdParams,
+        setGetSubmissionDetailsByIdParams ] = useState<IGetSubmissionDetailsByIdUrlParams | null>(null);
     const {
         get: getSubmissionDetails,
         data: apiSubmissionDetails,
-    } = useHttp(getSubmissionDetailsByIdUrl as UrlType);
+    } = useHttp(getSubmissionDetailsByIdUrl as UrlType, getSubmissionDetailsByIdParams);
 
+    const [
+        submissionResultsByProblemUrlParams,
+        setSubmissionResultsByProblemUrlParams ] =
+        useState<IGetSubmissionResultsByProblemUrlParams | null>();
     const {
         get: getProblemResultsRequest,
         data: apiProblemResults,
-    } = useHttp(getSubmissionResultsByProblemUrl as UrlType, null);
+    } = useHttp(getSubmissionResultsByProblemUrl as UrlType, submissionResultsByProblemUrlParams);
 
-    const getDetails = useCallback(async () => {
-        if (isNil(currentSubmissionId)) {
+    const getDetails = useCallback(async (submissionId: number) => {
+        if (isNil(submissionId)) {
             return;
         }
 
-        startLoading();
-        await getSubmissionDetails();
-        stopLoading();
-    }, [ currentSubmissionId, getSubmissionDetails, startLoading, stopLoading ]);
+        setGetSubmissionDetailsByIdParams({ submissionId } as IGetSubmissionDetailsByIdUrlParams);
+    }, []);
```

Comment:
I suggest we don't change this here as its a general issue. Since our update with links being functions most "get..." functions have been converted to functions that setParameters and dont return anything,